craft_providers/lxd/lxc.py

```diff
@@ -1218,3 +1218,51 @@ def is_pro_enabled(
                 brief=f"Failed to run `pro` command on {instance_name!r}.",
                 details=errors.details_from_called_process_error(error),
             ) from error
+
+    def enable_pro_service(
```

Comment:
From my understanding, we won't need a `disable` method for the purpose of building Pro-enabed rocks, because of the following reasons:
- we are not going to auto-clean after finishing a pack or lifecycle command, so there will be no auto-detachment or auto-disable
- if `rockcraft clean` is run, the whole managed instance will be deleted anyway
- if `--pro` was passed to a subsequent lifecycle command with a different service(s), rockcraft will throw an error and stop (so it won't disabled the old service(s) and enable the new one(s))